private/tools/java/com/github/bazelbuild/rules_jvm_external/resolver/gradle/GradleProject.java

```diff
@@ -0,0 +1,108 @@
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardOpenOption;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/**
+ * Used for building a fake gradle project with dependencies we have
+ * to run a custom gradle task to resolve the dependencies
+ */
+public class GradleProject implements AutoCloseable  {
+
+    private final Path projectDir;
+    private final Path gradleCacheDir;
+    private ProjectConnection connection;
+    private final Path gradleJavaHome;
+    private final EventListener eventListener;
+    private final Path initScript;
+
+    public GradleProject(Path projectDir, Path gradleCacheDir, Path gradleJavaHome, Path initScript, EventListener eventListener) {
+        this.projectDir = projectDir;
+        this.gradleCacheDir = gradleCacheDir;
+        this.gradleJavaHome = gradleJavaHome;
+        this.eventListener = eventListener;
+        this.initScript = initScript;
+    }
+
+    public void setupProject() throws IOException {
+        Files.createDirectories(projectDir);
+
+        Files.writeString(
+                projectDir.resolve("settings.gradle"),
+                "rootProject.name = 'rules_jvm_external'\n",
+                StandardOpenOption.CREATE, StandardOpenOption.TRUNCATE_EXISTING
+        );
+    }
+
+    public void connect(Path gradlePath) throws IOException {
+        System.setProperty("gradle.user.home", gradleCacheDir.toAbsolutePath().toString());
+        System.setProperty("org.gradle.parallel", "true");
+        connection = GradleConnector.newConnector()
+                .forProjectDirectory(projectDir.toFile())
+                .useInstallation(gradlePath.toFile())
+                .connect();
+    }
+
+    /**
+     * Triggers dependency resolution by running the custom task to resolve gradle dependencies
+     */
+    public GradleDependencyModel resolveDependencies(Map<String, String> gradleProperties) {
+        if (connection == null) {
+            throw new IllegalStateException("Gradle connection not established. Call connect() first.");
+        }
+
+        // This allows us to pass sensitive information like repository credentials without
+        // leaking it into the actual build file
+        List<String> arguments = gradleProperties.entrySet().stream()
+                .map(entry -> "-P" + entry.getKey() + "=" + entry.getValue())
+                .collect(Collectors.toList());
+        if(this.initScript != null) {
+            arguments.add("--init-script=" + this.initScript);
+        }
+
+    return connection
+        .model(GradleDependencyModel.class)
+        .addProgressListener(new GradleProgressListener(eventListener))
+        .setStandardError(System.err)
+        .withArguments(arguments)
+      //  .setJvmArguments("-agentlib:jdwp=transport=dt_socket,server=y,suspend=y,address=*:5005") // uncomment if you want to debug the plugin itself
```

Comment:
This can be uncommented and be used with `bazel run` to then debug the plugin itself (as it runs with the gradle daemon and not the bazel process). Rather needed to debug some complex issues.